framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -195,6 +195,11 @@ public AnnotatedTypeMirror visitMemberSelect(MemberSelectTree node,
             if (t instanceof AnnotatedDeclaredType || t instanceof AnnotatedArrayType || t instanceof AnnotatedTypeVariable) {
                 return AnnotatedTypes.asMemberOf(f.types, f, t, elt).asUse();
             }
+        } else {
```

Comment:
The comment above (regarding int.class) should be removed - it's in the wrong place and is superseded by the comment you added. The two comments about int.class and int[].class near the end of the file are vague and misleading - presumably visitPrimitiveType and visitArrayType are called after visitMemberSelect, which is not what the comments seem to imply. Your changes look good to me. Thanks.